docs/concepts/storage/volumes.md

```diff
@@ -990,16 +958,50 @@ several media types.
 ## Out-of-Tree Volume Plugins
 In addition to the previously listed volume types, storage vendors may create
 custom plugins without adding it to the Kubernetes repository. This can be
-achieved by using the `FlexVolume` plugin.
+achieved by using either the `CSI` plugin or the `FlexVolume` plugin.
+
+For storage vendors looking to create an out-of-tree volume plugin, [please refer to this FAQ](https://github.com/kubernetes/community/blob/master/sig-storage/volume-plugin-faq.md) for choosing between the plugin options.
 
+### CSI
+
+CSI stands for [Container Storage Interface](https://github.com/container-storage-interface/spec/blob/master/spec.md),
+a specification attempting to establish an industry standard interface that
+Container Orchestration Systems (COs) can use to expose arbitrary storage systems
+to their container workloads.
+For more information about the details, please check the
+[design proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md).
+
+<!-- TODO: add link to the kubernetes extension section -->
+The `csi` volume type is an in-tree CSI volume plugin for Pods to interact
+with external CSI volume drivers running on the same node.
+After having deployed a CSI compatible volume driver, users can use `csi` as the
+volume type to mount the storage provided by the driver.
+
+CSI persistent volume support is introduced in Kubernetes v1.9 as an alpha feature
+which has to be explicitly enabled by the cluster administrator. In other words,
+the cluster administrator needs to add "`CSIPersistentVolume=true`" to the
+"`--feature-gates=`" flag for the apiserver, the controller-manager and the kubelet
+components.
+
+A CSI persistent volume has the following fields for users to specify:
```

Comment:
"has the following fields" - does this mean "requires the following fields", which are specified by the user? And is it the end user making a manifest that will specify this, or a cluster adminstrator or operator that's running the cluster that defines these?

Reply:
Only the first two fields are required. In the case where the PersistentVolume is manually provisioned, the storage administrator (typically the cluster administrator) specify these fields. Will clarify